src/scenario.jl

```diff
@@ -401,11 +395,31 @@ NamedTuple of collated results
 - `bleaching_mortality` : Array, Log of mortalities caused by bleaching
 - `coral_dhw_log` : Array, Log of DHW tolerances / adaptation over time (only logged in debug mode)
 """
-function run_model(domain::Domain, param_set::DataFrameRow, functional_groups::Vector{Vector{FunctionalGroup}}, max_projected_cover::Vector{Float64})::NamedTuple
+function run_model(domain::Domain, param_set::Union{DataFrameRow,YAXArray})::NamedTuple
+    n_locs::Int64 = domain.coral_growth.n_locs
+    n_sizes::Int64 = domain.coral_growth.n_sizes
+    n_groups::Int64 = domain.coral_growth.n_groups
+    _bin_edges::Matrix{Float64} = bin_edges()
+    functional_groups = [
+        FunctionalGroup.(
+            eachrow(_bin_edges[:, 1:end-1]),
+            eachrow(_bin_edges[:, 2:end]),
+            eachrow(zeros(n_groups, n_sizes))
+        ) for _ in 1:n_locs
+    ]
+    return run_model(domain, param_set, functional_groups)
+end
+function run_model(
+    domain::Domain,
+    param_set::DataFrameRow,
+    functional_groups::Vector{Vector{FunctionalGroup}}
+)::NamedTuple
     ps = DataCube(Vector(param_set); factors=names(param_set))
-    return run_model(domain, ps, functional_groups, max_projected_cover)
+    return run_model(domain, ps, functional_groups)
 end
-function run_model(domain::Domain, param_set::YAXArray, functional_groups::Vector{Vector{FunctionalGroup}}, max_projected_cover::Vector{Float64})::NamedTuple
+function run_model(
+    domain::Domain, param_set::YAXArray, functional_groups::Vector{Vector{FunctionalGroup}}
```

Comment:
Is there a reason why `functional_groups` cannot be created inside `run_model()`?

It would simplify the chain of functions being called.

Reply:
They are being created outside so we can use as a cache.